flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -0,0 +1,202 @@
+							"If it is too small, may cause intermediate merging. But if it is too large, " +
+							"it will cause too many files opened at the same time, consume memory and lead to random reading.");
+
+	public static final ConfigOption<Boolean> SQL_EXEC_SORT_ASYNC_MERGE_ENABLED =
+			key("sql.exec.sort.async-merge.enabled")
+					.defaultValue(true)
+					.withDescription("Whether to asynchronously merge sorted spill files.");
+
+	// ------------------------------------------------------------------------
+	//  Spill Options
+	// ------------------------------------------------------------------------
+	public static final ConfigOption<Boolean> SQL_EXEC_SPILL_COMPRESSION_ENABLED =
+			key("sql.exec.spill.compression.enabled")
+					.defaultValue(true)
+					.withDescription("Whether to compress spilled data. " +
+							"(Now include sort and hash agg and hash join)");
+
+	public static final ConfigOption<String> SQL_EXEC_SPILL_COMPRESSION_CODEC =
+			key("sql.exec.spill.compression.codec")
+					.defaultValue("lz4")
+					.withDescription("Use that compression codec to compress spilled file. " +
+							"Now we only support lz4.");
+
+	public static final ConfigOption<Integer> SQL_EXEC_SPILL_COMPRESSION_BLOCK_SIZE =
+			key("sql.exec.spill.compression.block-size")
+					.defaultValue(64 * 1024)
+					.withDescription("The buffer is to compress. The larger the buffer," +
+							" the better the compression ratio, but the more memory consumption.");
+
+	// ------------------------------------------------------------------------
+	//  Resource Options
+	// ------------------------------------------------------------------------
+
+	public static final ConfigOption<Integer> SQL_RESOURCE_DEFAULT_PARALLELISM =
+			key("sql.resource.default.parallelism")
+					.defaultValue(-1)
+					.withDescription("Default parallelism of job operators. If it is <= 0, use parallelism of StreamExecutionEnvironment(" +
+							"its default value is the num of cpu cores in the client host).");
+
+	public static final ConfigOption<Integer> SQL_RESOURCE_SOURCE_PARALLELISM =
+			key("sql.resource.source.parallelism")
+					.defaultValue(-1)
+					.withDescription("Sets source parallelism, if it is <= 0, use " + SQL_RESOURCE_DEFAULT_PARALLELISM.key() + " to set source parallelism.");
+
+	public static final ConfigOption<Integer> SQL_RESOURCE_SINK_PARALLELISM =
+			key("sql.resource.sink.parallelism")
+					.defaultValue(-1)
+					.withDescription("Sets sink parallelism, if it is <= 0, use " + SQL_RESOURCE_DEFAULT_PARALLELISM.key() + " to set sink parallelism.");
+
+	public static final ConfigOption<Integer> SQL_RESOURCE_EXTERNAL_BUFFER_MEM =
+			key("sql.resource.external-buffer.memory.mb")
+					.defaultValue(10)
+					.withDescription("Sets the externalBuffer memory size that is used in sortMergeJoin and overWindow.");
+
+	public static final ConfigOption<Integer> SQL_RESOURCE_HASH_AGG_TABLE_MEM =
+			key("sql.resource.hash-agg.table.memory.mb")
```

Comment:
The `.mb` syntax is deprecated. See `org.apache.flink.configuration.TaskManagerOptions#TASK_MANAGER_HEAP_MEMORY` vs `org.apache.flink.configuration.TaskManagerOptions#TASK_MANAGER_HEAP_MEMORY_MB`